src/main/java/com/minecolonies/coremod/client/gui/WindowHutGuardTower.java

```diff
@@ -9,255 +9,484 @@
+                }
+            });
+        }
+        else if (task.equals(GuardTask.GUARD))
+        {
+            patrolList.setDataProvider(new ScrollingList.DataProvider()
+            {
+                @Override
+                public int getElementCount()
+                {
+                    return 1;
+                }
+
+                @Override
+                public void updateElement(final int index, @NotNull final Pane rowPane)
+                {
+                    final BlockPos pos = building.getGuardPos();
+                    rowPane.findPaneOfTypeByID("position", Label.class).setLabelText(pos.getX() + " " + pos.getY() + " " + pos.getZ());
+                }
+            });
+        }
+
+        final ScrollingList mobsList = findPaneOfTypeByID(GUI_ELEMENT_LIST_MOBS, ScrollingList.class);
+        mobsList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return mobsToAttack.size();
+            }
+
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
+            {
+                final String name = mobsToAttack.get(index).getName();
 
-        registerButton(BUTTON_TASK_PATROL, this::switchTask);
-        registerButton(BUTTON_TASK_FOLLOW, this::switchTask);
-        registerButton(BUTTON_TASK_GUARD, this::switchTask);
-        registerButton(BUTTON_SET_TARGET, this::setTarget);
+                rowPane.findPaneOfTypeByID(GUI_LIST_ELEMENT_NAME, Label.class).setLabelText(name);
 
-        buttonTaskPatrol = this.findPaneOfTypeByID(BUTTON_TASK_PATROL, Button.class);
-        buttonTaskFollow = this.findPaneOfTypeByID(BUTTON_TASK_FOLLOW, Button.class);
-        buttonTaskGuard = this.findPaneOfTypeByID(BUTTON_TASK_GUARD, Button.class);
+                final Button switchButton = rowPane.findPaneOfTypeByID(GUI_LIST_BUTTON_SWITCH, Button.class);
 
-        buttonSetTarget = this.findPaneOfTypeByID(BUTTON_SET_TARGET, Button.class);
+                if (mobsToAttack.get(index).hasAttack())
+                {
+                    switchButton.setLabel(GUI_SWITCH_ON);
+                }
+                else
+                {
+                    switchButton.setLabel(GUI_SWITCH_OFF);
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onButtonClicked(@NotNull final Button button)
+    {
+        final Pane currentPane = findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).getCurrentView();
+        if (currentPane != null)
+        {
+            final Button buttonNextPage = findPaneOfTypeByID(GUI_BUTTON_NEXT_PAGE, Button.class);
+            final Button buttonPrevPage = findPaneOfTypeByID(GUI_BUTTON_PREV_PAGE, Button.class);
+            switch (button.getID())
+            {
+                case GUI_BUTTON_NEXT_PAGE:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 11 to at most 10, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)